pgx/src/datum/numeric.rs

```diff
@@ -7,218 +7,225 @@ All rights reserved.
+            let zero: AnyNumeric = 0.try_into().unwrap();
+            if sign < zero {
+                return Sign::Negative;
+            } else if sign > zero {
+                return Sign::Positive;
+            } else {
+                return Sign::Zero;
+            }
+        }
+    }
+
+    /// The absolute value of this [`AnyNumeric`]
+    pub fn abs(&self) -> Self {
+        unsafe { direct_function_call(pg_sys::numeric_abs, vec![self.as_datum()]).unwrap() }
     }
-}
 
-impl From<u32> for Numeric {
-    fn from(val: u32) -> Self {
-        Numeric(val.to_string())
+    /// Compute the logarithm of this [`AnyNumeric`] in the given base
+    pub fn log(&self, base: AnyNumeric) -> Self {
+        unsafe {
+            direct_function_call(pg_sys::numeric_log, vec![self.as_datum(), base.as_datum()])
+                .unwrap()
+        }
     }
-}
 
-impl From<u64> for Numeric {
-    fn from(val: u64) -> Self {
-        Numeric(val.to_string())
+    /// Raise `e` to the power of `x`
+    pub fn exp(x: &AnyNumeric) -> Self {
+        unsafe { direct_function_call(pg_sys::numeric_exp, vec![x.as_datum()]).unwrap() }
     }
-}
 
-impl From<f32> for Numeric {
-    fn from(val: f32) -> Self {
-        Numeric(val.to_string())
+    /// Compute the square root of this [`AnyNumeric`]
+    pub fn sqrt(&self) -> Self {
+        unsafe { direct_function_call(pg_sys::numeric_sqrt, vec![self.as_datum()]).unwrap() }
     }
-}
 
-impl From<f64> for Numeric {
-    fn from(val: f64) -> Self {
-        Numeric(val.to_string())
+    /// Return the smallest integer greater than or equal to this [`AnyNumeric`]
+    pub fn ceil(&self) -> Self {
+        unsafe { direct_function_call(pg_sys::numeric_ceil, vec![self.as_datum()]).unwrap() }
     }
-}
 
-impl FromDatum for Numeric {
-    unsafe fn from_polymorphic_datum(
-        datum: pg_sys::Datum,
-        is_null: bool,
-        _typoid: u32,
-    ) -> Option<Self>
-    where
-        Self: Sized,
-    {
-        if is_null {
-            None
-        } else {
-            let cstr =
-                direct_function_call::<&std::ffi::CStr>(pg_sys::numeric_out, vec![Some(datum)])
-                    .expect("numeric_out returned null");
-            Some(Numeric(cstr.to_str().unwrap().into()))
-        }
+    /// Return the largest integer equal to or less than this [`AnyNumeric`]
+    pub fn floor(&self) -> Self {
+        unsafe { direct_function_call(pg_sys::numeric_floor, vec![self.as_datum()]).unwrap() }
     }
-}
 
-impl IntoDatum for Numeric {
-    fn into_datum(self) -> Option<pg_sys::Datum> {
-        let cstring =
-            std::ffi::CString::new(self.0).expect("failed to convert numeric string into CString");
-        let cstr = cstring.as_c_str();
+    /// Calculate the greatest common divisor of this an another [`AnyNumeric`]
+    #[cfg(not(any(feature = "pg11", feature = "pg12")))]
+    pub fn gcd<const P: u32, const S: u32>(&self, n: &AnyNumeric) -> AnyNumeric {
```

Comment:
I think the `P` and `S` params here are unnecessary.

Reply:
Good catch.  Looks like that was leftover from a previous iteration I was working on.